### PR TITLE
Fix analyzers warnings and suggestions

### PR DESCRIPTION
**Pull request by goremykin**
**Changes**
This PR is mostly about cleanup and fixing analyzers warnings.
Some changes may seem controversial or a matter of taste, so all changes are grouped by type of warning/suggestion (see commit names) to make it easy to rollback any you don't like.

### CODE REVIEW

File: tests/Jellyfin.Model.Tests/Dlna/StreamBuilderTests.cs [hunk 1]
```diff
@@ -162,7 +162,7 @@ public class StreamBuilderTests
         [InlineData("Tizen4-4K-5.1", "mkv-vp9-aac-srt-2600k", PlayMethod.DirectPlay)]
         [InlineData("Tizen4-4K-5.1", "mkv-vp9-ac3-srt-2600k", PlayMethod.DirectPlay)]
         [InlineData("Tizen4-4K-5.1", "mkv-vp9-vorbis-vtt-2600k", PlayMethod.DirectPlay)]
-        public async Task BuildVideoItemSimple(string deviceName, string mediaSource, PlayMethod? playMethod, TranscodeReason why = (TranscodeReason)0, string transcodeMode = "DirectStream", string transcodeProtocol = "")
+        public async Task BuildVideoItemSimple(string deviceName, string mediaSource, PlayMethod? playMethod, TranscodeReason why = 0, string transcodeMode = "DirectStream", string transcodeProtocol = "")
```
Comment (Bond-009) [inline tests/Jellyfin.Model.Tests/Dlna/StreamBuilderTests.cs:165]: Maybe `default` would be better than `0`?

**Review by Bond-009 — APPROVED** [review]

File: tests/Jellyfin.Model.Tests/Dlna/StreamBuilderTests.cs (hunk 1 above)
Comment (goremykin) [reply tests/Jellyfin.Model.Tests/Dlna/StreamBuilderTests.cs:165]: @Bond-009, There is a chance to change the behavior of the code in this case as enum starts from 1

Comment (Bond-009) [reply tests/Jellyfin.Model.Tests/Dlna/StreamBuilderTests.cs:165]: `default` for enums should always be `0` AFAIK?

Comment (goremykin) [reply tests/Jellyfin.Model.Tests/Dlna/StreamBuilderTests.cs:165]: Except when explicitly specified :)
You can check the source code of the enum and make sure

Comment (Bond-009) [reply tests/Jellyfin.Model.Tests/Dlna/StreamBuilderTests.cs:165]: Please show me how/where, I can only find this:
> The default value of an enumeration type E is the value produced by expression (E)0, even if zero doesn't have the corresponding enum member.

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/enum

Comment (goremykin) [reply tests/Jellyfin.Model.Tests/Dlna/StreamBuilderTests.cs:165]: You are absolutely right. I thought it sets the first enum value (1), but it sets 0 even if there is such a value.

Comment (goremykin) [reply tests/Jellyfin.Model.Tests/Dlna/StreamBuilderTests.cs:165]: So let's use default in this case

Comment (goremykin) [reply tests/Jellyfin.Model.Tests/Dlna/StreamBuilderTests.cs]: @Bond-009, done

**Review by 1337joe — APPROVED** [review]

**Pull request merged**